src/main/java/seedu/address/logic/parser/FindAppointmentCommandParser.java

```diff
@@ -0,0 +1,103 @@
+package seedu.address.logic.parser;
+
+import static java.util.Objects.requireNonNull;
+import static seedu.address.logic.parser.CliSyntax.PREFIX_DOCTOR;
+import static seedu.address.logic.parser.CliSyntax.PREFIX_NAME;
+import static seedu.address.logic.parser.CliSyntax.PREFIX_TAG;
+import static seedu.address.logic.parser.CliSyntax.PREFIX_TIMESLOT_START;
+
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.stream.Stream;
+
+import seedu.address.commons.core.Messages;
+import seedu.address.logic.commands.FindAppointmentCommand;
+import seedu.address.logic.commands.FindPatientCommand;
+import seedu.address.logic.parser.exceptions.ParseException;
+import seedu.address.model.appointment.AppointmentContainsKeywordsPredicate;
+
+/**
+ * Parses input arguments and creates a new FindAppointmentCommand object
+ */
+public class FindAppointmentCommandParser implements Parser<FindAppointmentCommand> {
+
+    /**
+     * Parses the given {@code String} of arguments in the context of the FindAppointmentCommand
+     * and returns a FindAppointmentCommand object for execution.
+     * @throws ParseException if the user input does not conform the expected format
+     */
+    public FindAppointmentCommand parse(String args) throws ParseException {
+        String trimmedArgs = args.trim();
+
+        if (trimmedArgs.isEmpty()) {
+            throw new ParseException(
+                    String.format(Messages.MESSAGE_INVALID_COMMAND_FORMAT, FindPatientCommand.MESSAGE_USAGE));
+        }
+
+        ArgumentMultimap argMultimap = ArgumentTokenizer.tokenize(args, PREFIX_NAME, PREFIX_DOCTOR,
+                PREFIX_TIMESLOT_START, PREFIX_TAG);
+
+        if (!areAnyPrefixesPresent(argMultimap, PREFIX_NAME, PREFIX_DOCTOR,
+                PREFIX_TIMESLOT_START, PREFIX_TAG)) {
+            throw new ParseException(
+                    String.format(Messages.MESSAGE_INVALID_COMMAND_FORMAT, FindAppointmentCommand.MESSAGE_USAGE));
+        }
+
+        ArrayList<String> patientKeywords = new ArrayList<String>();
+        ArrayList<String> doctorKeywords = new ArrayList<String>();
+        ArrayList<String> timeStartKeywords = new ArrayList<String>();
+        ArrayList<String> tagKeywords = new ArrayList<String>();
```

Comment:
programming to an interface is better coding practice if you don't need to use any `ArrayList` specific methods: ie. 
`List<String> xxx = new ArrayList<>();` instead of `ArrayList<String>` on the LHS